src/libultra/io/pimgr.c

```diff
@@ -46,7 +46,7 @@ void osCreatePiManager(OSPri pri, OSMesgQueue* cmdQ, OSMesg* cmdBuf, s32 cmdMsgC
         __osPiDevMgr.piDmaCallback = __osPiRawStartDma;
         __osPiDevMgr.epiDmaCallback = __osEPiRawStartDma;
 
-        osCreateThread(&piThread, 0, __osDevMgrMain, (void*)&__osPiDevMgr, piStackThread + sizeof(piStackThread), pri);
+        osCreateThread(&piThread, 0, __osDevMgrMain, (void*)&__osPiDevMgr, STACK_TOP(piStackThread), pri);
```

Comment:
This bit doesn't look right, specifically passing  `(void*)&__osPiDevMgr` instead of `piStackThread`. this could break if those variables get reordered, right?

Reply:
That arg is the `arg` that will be passed to the thread entrypoint function, it's unrelated to the stack

Reply:
oh wait no, I'm dumb, I was confusing this function with `StackCheck_Init`, forget about it